src/MigrationTools.Clients.AzureDevops.Rest/Processors/AzureDevOpsPipelineProcessor.cs

```diff
@@ -359,6 +364,38 @@ private async Task<IEnumerable<Mapping>> CreateBuildPipelinesAsync(IEnumerable<M
             return mappings;
         }
 
+        private void MapServiceConnectionsInBuidDefinition(IEnumerable<ServiceConnection> sourceServiceConnections, IEnumerable<ServiceConnection> targetServiceConnections, BuildDefinition definitionToBeMigrated)
+        {
+            definitionToBeMigrated.Process.Phases.ForEach(p =>
+            {
+                p.Steps.ForEach(s =>
+                {
+                    var mappings = new List<KeyValuePair<string, string>>();
+                    s.Inputs.ForEach(i =>
+                    {
+                        if (Guid.TryParse(i.Value.ToString(), out Guid currentGuid))
+                        {
+                            var sourceServiceConnection = sourceServiceConnections.FirstOrDefault(ssc => ssc.Id == i.Value.ToString());
+                            if (sourceServiceConnection == null) return;
+                            var targetServiceConnection = targetServiceConnections.FirstOrDefault(tsc => tsc.Name == sourceServiceConnection.Name);
+                            if (targetServiceConnection == null) return;
+                            Log.LogInformation($"- Updating service connection {p.Name}/{s.DisplayName}/{i.Key} {sourceServiceConnection.Id} to {targetServiceConnection.Id}...");
+                            mappings.Add(new KeyValuePair<string, string>(i.Key, i.Value.ToString()));
```

Comment:
Shouldn't the targetServiceConnection Id be used here? I tried tested your code and without that change, the ID was not getting mapped

```suggestion
                            mappings.Add(new KeyValuePair<string, string>(i.Key, targetServiceConnection.Id.ToString()));
```